src/Strategy/Maven/Common.hs

```diff
@@ -28,6 +28,15 @@ data MavenDependency = MavenDependency
 mavenDependencyToDependency :: MavenDependency -> Dependency
 mavenDependencyToDependency MavenDependency{..} = dependency
 
+-- | Filter all submodules (including their dependencies) that are not in `includedSubmoduleSet`.
+--
+--   We use `reachableSuccessorsWithCondition` to retrieve a submodule's dependencies so long as
+--   the dependency is not a submodule itself. This is important in the case that a dependency belongs to multiple submodules
```

Comment:
[nit] "so long as the dependency is not also a submodule in the project"

It makes it more obvious what the relationship between the submodules is, but this could also be less obvious to me just be because I'm not as familiar with this code as you are right now. I'll defer to your judgement whether the suggestion actually provides clarity.